### ISSUE TITLE
Major security flaw - loadFromCookie

### ISSUE THREAD

**Issue by PatrickG** [pocketbase/js-sdk#85]
There seems to be a major security flaw when using `client.authStore.loadFromCookie()`.
The cookie value is clear text and can easily be changed.
`loadFromCookie()` set the model data from the cookie value without validating anything (except that the token is not expired).

That means you can never trust `client.authStore.model` even if `client.authStore.valid` returns `true`.

I think it would be far better to use a jwt with the desired model data as the cookie value and do really verify the token (not just the expire date).

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what is the issue here.

The SDK is a client library. It doesn't matter what `client.authStore.isValid` is because the server has the correct security checks in place. If you want to make sure that client side your token is valid, you can send a `client.collection("users").refresh()` call first.

**Comment by ganigeorgiev** [comment]
You can find an example nodejs integration in https://github.com/pocketbase/pocketbase/discussions/690#discussioncomment-3782980

(the example is for v0.7.x and the only change compared to v0.8.x is that `req.pbClient.users.refresh();` should be `req.pbClient.collection("users").refresh();`)

**Comment by PatrickG** [comment]
I'm talking about this: [SSR integration](https://github.com/pocketbase/js-sdk#ssr-integration)

You can not trust `client.authStore.model`.

For example, checking if [a user should be able to see a specific page](https://github.com/huntabyte/projecthunt/blob/7dd5480824625c087e02d0d80efbf7a0fa199266/apps/web/src/routes/%2Blayout.server.ts), [Using `client.authStore.model.id` as value in the database](https://github.com/huntabyte/projecthunt/blob/2044e24a042dcfad8eaf2ae70f18151f2dcbe9d1/apps/web/src/routes/projects/new/%2Bpage.server.ts) or [checking if a record belongs to the current user](https://github.com/huntabyte/projecthunt/blob/a0f96c149c4df75fa34c1238b3faa96656057032/apps/web/src/routes/projects/%5BprojectId%5D/edit/%2Blayout.server.ts).

**Comment by ganigeorgiev** [comment]
@PatrickG Once again. It doesn't matter what is the value of the loaded cookie, because when making request to the PocketBase server it has the correct security checks in place. The cookie only loads the JWT token in the SDK instance so that it can be passed as `Authorization: TOKEN` to the consequent requests.

If you are using only `client.authStore.isValid` as a security checks, then this is wrong. This is not a session cookie or anything like that since the check doesn't send a request to the server. I'll make sure sometimes later this week to update the example and to clarify it to avoid the confusion.

**Comment by PatrickG** [comment]
I know that the token is used to communicate with the pocketbase api.
But if whole `model` in the cookie is useless.

**Comment by ganigeorgiev** [comment]
The `model` in the cookie is passed for convenience so that users don't have to keep track of the auth instance on their own. If you want valid and up-to-date user state, call `pb.collection("users").refresh()`. With v0.8.0 you can also attach to realtime changes of your user model:

```js
pb.collection("users").subscribe("USER_ID", (e) => {
    console.log(e.record)
})
```

**Comment by PatrickG** [comment]
It seems like this lines from the [discussion](https://github.com/pocketbase/pocketbase/discussions/690#discussioncomment-3782980) you've linked are missing in the [SSR Integration docs](https://github.com/pocketbase/js-sdk#ssr-integration).

```js
    try {
        // get an up-to-date auth state by refreshing the loaded auth model (if any)
        if (req.pbClient.authStore.isValid) {
            await req.pbClient.users.refresh();
        }
    } catch (_) {
        // clear the auth store on failed refresh
        req.pbClient.authStore.clear();
    }
```

**Comment by ganigeorgiev** [comment]
@PatrickG As I mentioned earlier I'll update the examples sometime later this week to state it more clearly, but keep in mind that the above is not required if you are communicating only with the PocketBase server. The issue here is that you rely on the `pb.authStore.isValid` state to determine access permissions in your node application.

The cookie load and export methods are just helpers to help you populate the store. Under the hood `pb.authStore.loadFromCookie()` is doing `pb.authStore.save(token, model)`, non of which is a server-side operation.

**Comment by ganigeorgiev** [comment]
@PatrickG I've created a PR in the tutorial repo - https://github.com/huntabyte/projecthunt/pull/4.

If you aware of other tutorials feel free to post them here or directly file a PR/issue to the authors.

**Comment by PatrickG** [comment]
> @PatrickG I've created a PR in the tutorial repo - [huntabyte/projecthunt#4](https://github.com/huntabyte/projecthunt/pull/4).
> 
> If you aware of other tutorials feel free to post them here or directly file a PR/issue in the authors repo.

Thank you.
I've created the same pr for another repo.
I hope it is okay that i copied your text for this :)

**Comment by ganigeorgiev** [comment]
@PatrickG please note that in the [kitbase repo](https://github.com/kevmodrome/kitbase/pull/1) they are using PocketBase SDK v0.7.x and the API syntax is a little different. It should be:

```js
try {
  // get an up-to-date auth store state by veryfing and refreshing the loaded auth model (if any)
  if (event.locals.pb.authStore.isValid) {
    await event.locals.pb.users.authRefresh();
  }
} catch (_) {
  // clear the auth store on failed refresh
  event.locals.pb.authStore.clear();
}
```

**Comment by PatrickG** [comment]
Alright, fixed my pr.

**Comment by ganigeorgiev** [comment]
I've updated the examples and I've added a note in the `loadFromCookie` jsdoc that you may need to call `authRefresh()` in order to trust the `isValid` call. It is published as a new [v0.8.1 release](https://github.com/pocketbase/js-sdk/releases/tag/v0.8.1) so that the docs changes can be available on autocomplete and in the npm package readme.
